### DISCUSSION TITLE
Is there any way to use sysctl parameters for kafka pods?

### DISCUSSION THREAD

**Discussion by Amet13** [strimzi/strimzi-kafka-operator#8036] [General]
**Suggestion / Problem**
We need to set some sysctl parameter for kafka pod, like:
```
sysctl -w vm.max_map_count=262144
```
Seems like we are not able to do that using [initConainer](https://github.com/strimzi/strimzi-kafka-operator/issues/2976) for now.

But there is some placeholder for sysctls in the [CRD](https://github.com/strimzi/strimzi-kafka-operator/blob/main/helm-charts/helm3/strimzi-kafka-operator/crds/040-Crd-kafka.yaml#L900).

Is there any solution for using sysctl params for kafka cluster/pods?

**Comment by scholzj** [comment]
Not as far as I know. I think normally you change these parameters on the Kubernetes worker node.

**Reply by ghost** [reply]
@scholzj Do we intend to add the support for the same in future strimzi release , these kernel parameters can be updated as part of  the initContainer .

**Reply by scholzj** [reply]
I personally do not intend to add any support for that. As I said, AFAIK this should be set on the worker node. Playing with system parameters like this from init containers is from my experience unreliable and does not work in many environments anyway because of security.

**Comment by craiglservin** [comment]
We ended up using a DaemonSet to set `sysctl` values for broker pods, and having a separate node pool for brokers so we could ensure one broker pod per node.  I did not want to share the Linux buffer/page cache of the node between multiple broker pods.
